From 86b04cf3b539634926b191ddbb5aad38b4ee595a Mon Sep 17 00:00:00 2001
From: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Date: Wed, 17 Jul 2013 13:17:00 -0700
Subject: Dalvik: Getting Registerization Rejection in logcat with -Xjitverbose

BZ: 124253

This patch changes the ALOGV to ALOGD and protects it with the check to
-Xjitverbose being set.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: Ic09a4a47c8e8a5dfe80042d185dde1246148deb9
Orig-MCG-Change-Id: I480e55a4b998482a1742c83e452a032c33c4074a
Signed-off-by: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Reviewed-on: http://android.intel.com:8080/123672
Reviewed-by: Popov, Ivan G <ivan.g.popov@intel.com>
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/RegisterizationME.cpp |   11 +++++++----
 1 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/vm/compiler/RegisterizationME.cpp b/vm/compiler/RegisterizationME.cpp
index 76ceb7f..cbfff4a 100644
--- a/vm/compiler/RegisterizationME.cpp
+++ b/vm/compiler/RegisterizationME.cpp
@@ -524,9 +524,10 @@ void dvmCompilerRegisterize (CompilationUnit *cUnit, Pass *currentPass)
 /**
  * @brief Parse a BasicBlock of the loop
  * @param bb the BasicBlock of the loop
+ * @param verbose are we in verbose mode or not
  * @return whether the BasicBlock contains any opcodes we don't want to support for registerization
  */
-static bool parseBlock (BasicBlock *bb)
+static bool parseBlock (BasicBlock *bb, bool verbose)
 {
     //Go through the instructions
     for (MIR *mir = bb->firstMIRInsn; mir != 0; mir = mir->next)
@@ -720,8 +721,10 @@ static bool parseBlock (BasicBlock *bb)
 
             default:
                 //Not accepted, refuse the block
-                ALOGV ("Rejecting registerization due to %s",
-                        dvmCompilerGetDalvikDisassembly (& (mir->dalvikInsn), NULL));
+                if (verbose == true)
+                {
+                    ALOGD ("Rejecting registerization due to %s", dvmCompilerGetDalvikDisassembly (& (mir->dalvikInsn), NULL));
+                }
                 return false;
         }
     }
@@ -771,7 +774,7 @@ static bool checkLoop (const CompilationUnit *cUnit, LoopInformation *info, void
         }
 
         //If we don't accept this one, refuse everything
-        if (parseBlock (bb) == false)
+        if (parseBlock (bb, cUnit->printMe) == false)
         {
             return false;
         }
-- 
1.7.4.1

